zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeperMain.java

```diff
@@ -283,44 +288,17 @@ void run() throws IOException, InterruptedException {
         if (cl.getCommand() == null) {
             System.out.println("Welcome to ZooKeeper!");
 
-            boolean jlinemissing = false;
-            // only use jline if it's in the classpath
-            try {
-                Class<?> consoleC = Class.forName("jline.console.ConsoleReader");
-                Class<?> completorC = Class.forName("org.apache.zookeeper.JLineZNodeCompleter");
-
-                System.out.println("JLine support is enabled");
+            Terminal terminal = TerminalBuilder.builder().system(true).build();
+            Completer znodeCompleter = new JLineZNodeCompleter(zk);
+            String prompt = "[zkCli] ";
 
-                Object console = consoleC.getConstructor().newInstance();
+            LineReader lineReader = LineReaderBuilder.builder().terminal(terminal).completer(znodeCompleter).build();
```

Comment:
Great, the new api looks more concise!

Reply:
Thank you very much @maoling . I have learned a lot of useful knowledge from your code review comments.